crates/biome_js_analyze/src/lint/nursery/use_qwik_method_usage.rs

```diff
@@ -0,0 +1,170 @@
+    /// });
+    ///
+    /// export const useCounter = () => {
+    ///   const count = useSignal(0);
+    ///   return count;
+    /// };
+    /// ```
+    pub UseQwikMethodUsage {
+        version: "next",
+        name: "useQwikMethodUsage",
+        language: "jsx",
+        sources: &[RuleSource::EslintQwik("use-method-usage").same()],
+        recommended: true,
+        severity: Severity::Error,
+        domains: &[RuleDomain::Qwik],
+    }
+}
+
+impl Rule for UseQwikMethodUsage {
+    type Query = Ast<JsCallExpression>;
+    type State = biome_rowan::TextRange;
+    type Signals = Option<Self::State>;
+    type Options = UseQwikMethodUsageOptions;
+
+    fn run(ctx: &biome_analyze::context::RuleContext<Self>) -> Self::Signals {
+        let call = ctx.query();
+        let is_hook = is_qwik_hook(call)?;
+        if !is_hook {
+            return None;
+        }
+        let is_valid_context = is_inside_component_or_hook(call).unwrap_or(false)
+            || is_in_named_function(call).unwrap_or(false);
+        if is_valid_context {
+            None
+        } else {
+            Some(call.range())
+        }
+    }
+
+    fn diagnostic(
+        _: &biome_analyze::context::RuleContext<Self>,
+        range: &Self::State,
+    ) -> Option<RuleDiagnostic> {
+        RuleDiagnostic::new(
+            rule_category!(),
+            range,
+            markup! {
+                "Qwik hook detected outside of allowed scope"
+            },
+        )
+            .note(markup! {
+            "Qwik's reactive hooks (functions starting with  "<Emphasis>"use*"</Emphasis>" followed by uppercase letter) must be:"
+            "\n- Used exclusively within `component$` functions"
+            "\n- Or encapsulated within other valid Qwik hooks"
+        })
+            .note(markup! {
+            "Check the "<Hyperlink href="https://qwik.dev/docs/components/state/#use-methods">"Qwik documentation"</Hyperlink>"."
+        })
+            .into()
+    }
+}
+
+fn is_qwik_hook(call: &JsCallExpression) -> Option<bool> {
+    let binding = call
+        .callee()
+        .ok()?
+        .as_js_reference_identifier()?
+        .value_token()
+        .ok()?;
+    let name = binding.text();
+    Some(name.starts_with("use") && name.chars().nth(3).is_some_and(|c| c.is_uppercase()))
+}
+
+fn is_inside_component_or_hook(call: &JsCallExpression) -> Option<bool> {
+    let outer_call = call
+        .syntax()
+        .ancestors()
+        .find_map(AnyJsFunction::cast)
+        .and_then(|function| {
+            function
+                .syntax()
+                .ancestors()
+                .find_map(JsCallExpression::cast)
+        });
+
+    outer_call.and_then(|call_expr| {
+        call_expr
+            .callee()
+            .ok()
+            .and_then(|callee| callee.as_js_reference_identifier())
+            .and_then(|ident| ident.value_token().ok())
+            .map(|token| {
+                let name = token.text();
```

Comment:
```suggestion
                let name = token.text_trimmed();
```

This is a bug. It will fail if there are spaces or comments between `component$`